target/linux/rtl838x/files-5.4/drivers/net/ethernet/rtl838x_eth.c

```diff
@@ -0,0 +1,1205 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * linux/drivers/net/ethernet/rtl838x_eth.c
+ * Copyright (C) 2020 B. Koblitz
+ */
+
+#include <linux/dma-mapping.h>
+#include <linux/etherdevice.h>
+#include <linux/interrupt.h>
+#include <linux/io.h>
+#include <linux/platform_device.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+#include <linux/of.h>
+#include <linux/of_net.h>
+#include <linux/of_mdio.h>
+#include <linux/module.h>
+#include <linux/phylink.h>
+#include <net/dsa.h>
+
+#include <asm/mach-rtl838x/mach-rtl838x.h>
+#include "rtl838x_eth.h"
+
+extern struct rtl838x_soc_info soc_info;
```

Comment:
From checkpatch.pl:
`WARNING: externs should be avoided in .c files`

Reply:
I have seen similar usage of external structures containing SoC configuration values elsewhere in the kernel.